packages/flame/lib/src/components/spawn_component.dart

```diff
@@ -58,13 +72,37 @@ class SpawnComponent extends Component {
         ),
         _period = minPeriod +
             (random ?? randomFallback).nextDouble() * (maxPeriod - minPeriod),
+        multiFactory = multiFactory ?? _wrapFactory(factory!),
         _random = random ?? randomFallback;
 
+  /// The function used to create a new component to spawn.
+  ///
+  /// [amount] is the amount of components that the [SpawnComponent] has spawned
+  /// so far.
+  ///
+  /// Be aware: internally the component uses a factory that creates a list of
+  /// components.
+  /// If you have set such a factory it was wrapped to create a list. The factory
+  /// getter wrapps it again to return the first element of the list and fails
+  /// when the list is empty!
```

Comment:
Maybe this could be made as an assertion instead?

Reply:
Well, this is a weird usage of the API and would fail when testing it right away, but I could add a test and throw a more meaningful exception than out of bounds

Reply:
It's better to have it as an assertion than checking and throwing another exception, since the check then doesn't have to be done in production.